src/Microsoft.DotNet.Darc/src/Darc/Operations/GetGoalOperation.cs

```diff
@@ -0,0 +1,55 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.Darc.Helpers;
+using Microsoft.DotNet.Darc.Options;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Net;
+using System.Threading.Tasks;
+using System.Linq;
+using Microsoft.DotNet.Maestro.Client.Models;
+using Microsoft.DotNet.Maestro.Client;
+
+namespace Microsoft.DotNet.Darc.Operations
+{
+    internal class GetGoalOperation : Operation
+    {
+        private GetGoalCommandLineOptions _options;
+        public GetGoalOperation(GetGoalCommandLineOptions options)
+            : base(options)
+        {
+            _options = options;
+        }
+
+        /// <summary>
+        /// Retrieve Goal in minutes for Definition in a Channel.
+        /// </summary>
+        /// <returns>Process exit code.</returns>
+        public override async Task<int> ExecuteAsync()
+        {
+            try
+            {
+                IRemote remote = RemoteFactory.GetBarOnlyRemote(_options, Logger);
+                Goal goalInfo = await remote.GetGoalAsync(_options.Channel, _options.DefinitionId);
+                if(goalInfo != null)
```

Comment:
This is an... odd check. If it _is_ null we just... do nothing? Does that make any sense?  We should probably either just assume it's null and not check (since we own the other side of this pipe, and null isn't a valid answer), or have some sort of answer for if it's null.  Silently do nothing but also return success is very unexpected behavior.

Reply:
We are taking care of the exceptions in the back-end API part. That is the style we are using. Normally we don't check for null here, that is already taken care of. I will remove the null check here, then it will be better